### PR TITLE
changelog for v0.10.0

### PR DESCRIPTION
**Pull request by jasonwilliams**
changelog for v0.10.0

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,112 @@
...
+# 0.10.0 (2020-09-29) - New Lexer & Test 262 Harness
+
+Feature Enhancements:
+
+- [FEATURE #559](https://github.com/boa-dev/boa/pull/559):
+  New Lexer (@Lan2u @HalidOdat @Razican)
+- [FEATURE #567](https://github.com/boa-dev/boa/pull/567):
+  Add ECMAScript test suite (test262) (@Razican)
+- [FEATURE #524](https://github.com/boa-dev/boa/issues/524):
+  Implement remaining `Math` methods (@mr-rodgers)
```
Comment (joshwd36) [inline CHANGELOG.md:12]:
```suggestion
- [FEATURE #524](https://github.com/boa-dev/boa/pull/525):
  Implement remaining `Math` methods (@mr-rodgers)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,112 @@
...
+- [FEATURE #536](https://github.com/boa-dev/boa/pull/536):
+  Implement `SyntaxError` (@HalidOdat)
+- [FEATURE #543](https://github.com/boa-dev/boa/pull/543):
+  Implements Object.create builtin method (@croraf)
+- [FEATURE #492](https://github.com/boa-dev/boa/pull/492):
+  Switch to [rustyline](https://github.com/kkawakam/rustyline) for the CLI (@IovoslavIovchev & @Razican)
+- [FEATURE #555](https://github.com/boa-dev/boa/pull/555):
+  Implement Array.prototype.reduce (@benjaminflin)
+- [FEATURE #550](https://github.com/boa-dev/boa/pull/550):
+  Initial implementation of Map() (@HalidOdat)
```
Comment (joshwd36) [inline CHANGELOG.md:24]:
```suggestion
- [FEATURE #550](https://github.com/boa-dev/boa/pull/550):
  Initial implementation of Map() (@joshwd36 & @HalidOdat)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,112 @@
...
+- [FEATURE #579](https://github.com/boa-dev/boa/pull/579):
+  Implement Array.prototype.reduceRight (@benjaminflin)
+- [FEATURE #585](https://github.com/boa-dev/boa/issues/585):
+  Implement Well-Known Symbols (@joshwd36)
```
Comment (joshwd36) [inline CHANGELOG.md:28]:
```suggestion
- [FEATURE #585](https://github.com/boa-dev/boa/pull/587):
  Implement Well-Known Symbols (@joshwd36)
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,112 @@
...
+- [FEATURE #589](https://github.com/boa-dev/boa/pull/589):
+  Implement the comma operator (@KashParty)
+- [FEATURE #341](https://github.com/boa-dev/boa/issues/341):
+  Ability to create multiline blocks in boa shell (@HalidOdat)
```
Comment (joshwd36) [inline CHANGELOG.md:32]:
```suggestion
- [FEATURE #341](https://github.com/boa-dev/boa/pull/590):
  Ability to create multiline blocks in boa shell (@HalidOdat)
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,112 @@
...
+- [FEATURE #252](https://github.com/boa-dev/boa/issues/252):
+  Implement `Date` (@jcdickinson)
```
Comment (joshwd36) [inline CHANGELOG.md:34]:
```suggestion
- [FEATURE #252](https://github.com/boa-dev/boa/pull/596):
  Implement `Date` (@jcdickinson)
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,112 @@
...
+Bug Fixes:
+
+- [BUG #405](https://github.com/boa-dev/boa/issues/405):
+  Fix json.stringify symbol handling (@n14little)
+- [BUG #520](https://github.com/boa-dev/boa/pull/520):
+  Fix all `Value` operations and add unsigned shift right (@HalidOdat)
+- [BUG #529](https://github.com/boa-dev/boa/pull/529):
+  Refactor exec/expression into exec/call and exec/new (@croraf)
+- [BUG #510](https://github.com/boa-dev/boa/issues/510):
+  [[Call]] calling an undefined method does not throw (@joshwd36)
```
Comment (joshwd36) [inline CHANGELOG.md:49]:
```suggestion
- [BUG #510](https://github.com/boa-dev/boa/pull/561):
  [[Call]] calling an undefined method does not throw (@joshwd36)
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,112 @@
...
+- [BUG #493](https://github.com/boa-dev/boa/pull/493):
+  Use correct exponential representation for rational values (@Tropid)
+- [BUG #572](https://github.com/boa-dev/boa/pull/572):
+  Spec Compliant `Number.prototype.toString()`, better `Number` object formating and `-0` (@HalidOdat)
+- [BUG #599](https://github.com/boa-dev/boa/pull/599):
+  Fixed `String.prototype.indexOf()` bug, when the search string is empty (@HalidOdat)
+- [BUG #615](https://github.com/boa-dev/boa/issues/615):
+  Fix abstract relational comparison operators
```
Comment (joshwd36) [inline CHANGELOG.md:57]:
```suggestion
- [BUG #615](https://github.com/boa-dev/boa/pull/617):
  Fix abstract relational comparison operators
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,112 @@
...
+- [BUG #608](https://github.com/boa-dev/boa/issues/608):
+  `Debug::fmt` Causes Causes a Stack Overflow (@jcdickinson)
```
Comment (joshwd36) [inline CHANGELOG.md:59]:
```suggestion
- [BUG #608](https://github.com/boa-dev/boa/pull/630):
  `Debug::fmt` Causes Causes a Stack Overflow (@jcdickinson)
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,112 @@
...
+- [BUG #532](https://github.com/boa-dev/boa/issues/532)
+  [builtins - Object] Object.getPrototypeOf returning incorrectly (@54k1)
+- [BUG #533](https://github.com/boa-dev/boa/issues/533)
+  [exec - function] function.prototype doesn't have own constructor property pointing to this function (@54k1)
+- [BUG #641](https://github.com/boa-dev/boa/issues/641)
+  Test new_instance_should_point_to_prototype is not checked correctly (@54k1)
+- [BUG #644](https://github.com/boa-dev/boa/issues/644)
+  `undefined` constants panic on execution (@jcdickinson)
```
Comment (joshwd36) [inline CHANGELOG.md:67]:
```suggestion
- [BUG #644](https://github.com/boa-dev/boa/pull/645)
  `undefined` constants panic on execution (@jcdickinson)
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,112 @@
...
+- [INTERNAL #544](https://github.com/boa-dev/boa/pull/544):
+  Removed `console`s dependency of `InternalState` (@HalidOdat)
+- [INTERNAL #556](https://github.com/boa-dev/boa/pull/556):
+  Added benchmark for goal symbol switching (@Razican)
+- [INTERNAL #578](https://github.com/boa-dev/boa/issues/578):
+  Extract `prototype` from internal slots (@HalidOdat)
+- [INTERNAL #553](https://github.com/boa-dev/boa/pull/553):
+  Refactor Property Descriptor flags (@HalidOdat)
+- [INTERNAL #582](https://github.com/boa-dev/boa/pull/582):
+  Make `String.prototype.repeat()` ECMAScript specification compliant (@HalidOdat)
```
Comment (Razican) [inline CHANGELOG.md:90]: I would say this is a fix.

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,5 +1,112 @@
...
+- [BUG #532](https://github.com/boa-dev/boa/issues/532)
+  [builtins - Object] Object.getPrototypeOf returning incorrectly (@54k1)
+- [BUG #533](https://github.com/boa-dev/boa/issues/533)
+  [exec - function] function.prototype doesn't have own constructor property pointing to this function (@54k1)
+- [BUG #641](https://github.com/boa-dev/boa/issues/641)
+  Test new_instance_should_point_to_prototype is not checked correctly (@54k1)
+- [BUG #644](https://github.com/boa-dev/boa/issues/644)
+  `undefined` constants panic on execution (@jcdickinson)
+- [BUG #631](https://github.com/boa-dev/boa/issues/631):
+  Unexpected result when applying typeof to undefined value (@jcdickinson)
```
Comment (joshwd36) [inline CHANGELOG.md:69]:
```suggestion
- [BUG #631](https://github.com/boa-dev/boa/pull/645):
  Unexpected result when applying typeof to undefined value (@jcdickinson)
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,5 +1,112 @@
...
+- [INTERNAL #582](https://github.com/boa-dev/boa/pull/582):
+  Make `String.prototype.repeat()` ECMAScript specification compliant (@HalidOdat)
+- [INTERNAL #373](https://github.com/boa-dev/boa/pull/373):
+  Introduce PropertyKey for field acces (@RageKnify)
+- [INTERNAL #592](https://github.com/boa-dev/boa/pull/592):
+  `RegExp` specialization (@HalidOdat)
+- [INTERNAL #597](https://github.com/boa-dev/boa/pull/597):
+  Made `String.prototype.indexOf` spec compliant. (@HalidOdat)
+- [INTERNAL #598](https://github.com/boa-dev/boa/pull/598):
+  Made `String.prototype.lastIndexOf()` spec compliant (@HalidOdat)
```
Comment (Razican) [inline CHANGELOG.md:98]: I also think these are fixes.

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,5 +1,112 @@
...
+- [BUG #533](https://github.com/boa-dev/boa/issues/533)
+  [exec - function] function.prototype doesn't have own constructor property pointing to this function (@54k1)
+- [BUG #641](https://github.com/boa-dev/boa/issues/641)
+  Test new_instance_should_point_to_prototype is not checked correctly (@54k1)
+- [BUG #644](https://github.com/boa-dev/boa/issues/644)
+  `undefined` constants panic on execution (@jcdickinson)
+- [BUG #631](https://github.com/boa-dev/boa/issues/631):
+  Unexpected result when applying typeof to undefined value (@jcdickinson)
+- [BUG #610](https://github.com/boa-dev/boa/issues/610):
+  Fix: String.prototype.replace substitutions (@RageKnify)
```
Comment (joshwd36) [inline CHANGELOG.md:71]:
```suggestion
- [BUG #610](https://github.com/boa-dev/boa/pull/629):
  Fix: String.prototype.replace substitutions (@RageKnify)
```

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,5 +1,112 @@
 # CHANGELOG
 
+# 0.10.0 (2020-09-29) - New Lexer & Test 262 Harness
+
+Feature Enhancements:
+
+- [FEATURE #559](https://github.com/boa-dev/boa/pull/559):
+  New Lexer (@Lan2u @HalidOdat @Razican)
```
Comment (Razican) [inline CHANGELOG.md:8]: This should probably be an internal enhancement, since it's not a new feature, right?

**Review by Razican — APPROVED** [review]
Looks good to me, I would just move some things from one section to another, as I mentioned.

File: CHANGELOG.md [hunk 15]
```diff
@@ -1,5 +1,112 @@
...
+Internal Improvements:
+
+- [INTERNAL #541](https://github.com/boa-dev/boa/pull/541):
+  Made all `Math` methods spec compliant (@HalidOdat)
+- [INTERNAL #544](https://github.com/boa-dev/boa/pull/544):
+  Removed `console`s dependency of `InternalState` (@HalidOdat)
+- [INTERNAL #556](https://github.com/boa-dev/boa/pull/556):
+  Added benchmark for goal symbol switching (@Razican)
+- [INTERNAL #578](https://github.com/boa-dev/boa/issues/578):
+  Extract `prototype` from internal slots (@HalidOdat)
```
Comment (joshwd36) [inline CHANGELOG.md:86]:
```suggestion
- [INTERNAL #578](https://github.com/boa-dev/boa/pull/580):
  Extract `prototype` from internal slots (@HalidOdat)
```

File: CHANGELOG.md [hunk 16]
```diff
@@ -1,5 +1,112 @@
...
+- [INTERNAL #592](https://github.com/boa-dev/boa/pull/592):
+  `RegExp` specialization (@HalidOdat)
+- [INTERNAL #597](https://github.com/boa-dev/boa/pull/597):
+  Made `String.prototype.indexOf` spec compliant. (@HalidOdat)
+- [INTERNAL #598](https://github.com/boa-dev/boa/pull/598):
+  Made `String.prototype.lastIndexOf()` spec compliant (@HalidOdat)
+- [INTERNAL #626](https://github.com/boa-dev/boa/pull/626):
+  Refactor `Function` (@HalidOdat @Razican)
+- [INTERNAL #564](https://github.com/boa-dev/boa/issues/564):
+  Add benchmarks for "uglified" JS (@neeldug)
```
Comment (joshwd36) [inline CHANGELOG.md:102]:
```suggestion
- [INTERNAL #564](https://github.com/boa-dev/boa/pull/581):
  Add benchmarks for "uglified" JS (@neeldug)
```

**Review by joshwd36 — COMMENTED** [review]
Looks good. I think it would be good to generally link to pull requests rather than issues, and there was one or two things not properly attributed (I think I got them all but there may be more)

**Comment by jasonwilliams** [comment]
I agree, i think from now on im going to link to PRs

**Comment by Razican** [comment]
This requires a re-base in order to fix the CI, now that we upgraded `criterion` and `critcmp`.

**Pull request merged**